src/redux/BookSlice.ts

```diff
@@ -0,0 +1,50 @@
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit'
+import { GoogleBook } from '../api/GoogleBook'
+import { Items, State } from '../types/GoogleBooks'
+import { RootState } from './Store'
+const initialState: State = {
+  searchInput: '',
+  loading: false,
+  error: false,
+  fetchedBooks: [],
+}
+const VOLUMES_SEAECH_PATH = '/volumes?q='
+export const GoogleBooks = createAsyncThunk(
+  'book/fetchBooks',
+  async (initalParameter: string) => {
+    const response = await GoogleBook.get<Items>(
+      VOLUMES_SEAECH_PATH + initalParameter
+    )
+    return response.data.items
+  }
+)
+export const BookSlice = createSlice({
+  name: 'book',
+  initialState,
+  reducers: {},
+  extraReducers: (builder) => {
+    builder
+      .addCase(GoogleBooks.pending, (state) => {
+        state.loading = true
+        state.fetchedBooks = []
+      })
+      .addCase(GoogleBooks.fulfilled, (state, action) => {
+        state.searchInput = action.meta.arg
```

Comment:
```ts
state.searchInput = action.meta.arg
```
I'm not sure what's happening here

Create a synchronous reducer for the searchInput state